articles/iot-hub/quickstart-device-streams-echo-csharp.md

```diff
@@ -104,19 +104,21 @@ dotnet build
 
 # Run the application
 # In Linux/MacOS
-dotnet run $ServiceConnectionString MyDevice
+dotnet run "ServiceConnectionString" "MyDevice"
 
 # In Windows
-dotnet run %ServiceConnectionString% MyDevice
+dotnet run ServiceConnectionString MyDevice
 ```
 
+*Note*: A timeout will occur if the device-side application does not respond in time.
+
 ### Run the device-side application
 
-Navigate to `device-streams-echo/device` directory in your unzipped project folder. You will need the following information handy:
+Navigate to `iot-hub/Quickstarts/device-streams-echo/device` directory in your unzipped project folder. You will need the following information handy:
 
 | Parameter name | Parameter value |
 |----------------|-----------------|
-| `DeviceConnectionString` | The connection string of the device you created earlier. |
+| `DeviceConnectionString` | The connection string of the device you created earlier. Replace DeviceConnectionString |
```

Comment:
Let's rephrase the whole sentence as: "Provide the device connection string of your IoT Hub."